lib/matplotlib/legend.py

```diff
@@ -333,6 +336,8 @@ def __init__(
         frameon=None,         # draw frame
         handler_map=None,
         title_fontproperties=None,  # properties for the legend title
+        *,
+        ncol=1  # synonym for ncols (backward compatibility)
```

Comment:
Can we do this?  I think you have to leave `ncol` where it was for backcompat?  We should prob move all of these to kwarg only...

Reply:
If someone uses this with a positional argument it will be picked up as `ncols`, so that should be fine. `ncols` will also come earlier when editors pop up the list of arguments, so I think this is really the way to go.

As #23166 is in, we are on the way to keyword only arguments.

Reply:
Oops yes I agree.